src/coprocessor/dag/expr/mod.rs

```diff
@@ -63,6 +67,24 @@ pub struct ScalarFunc {
     sig: ScalarFuncSig,
     children: Vec<Expression>,
     tp: FieldType,
+    cus_rng: RefCell<Option<CusRng>>,
```

Comment:
Could we use `RefCell<Option<XorShiftRng>>` instead?

Reply:
I’m not sure if I can implement debug and partial equal trait to xorshiftrng here. I can give it a try.

Reply:
There is an error: only traits defined in the current crate can be implemented for arbitrary types.
Seems we have to wrap XorShiftRng in CusRng and implement the needed traits for CusRng. @AndreMouche

Reply:
I think you can try to remove the `RefCell` from `CusRng` on the first step, then try to use RefCell<Option<XorShiftRng>> instead.  The following docs may help:
https://doc.rust-lang.org/std/cell/struct.RefCell.html
https://doc.rust-lang.org/std/fmt/trait.Debug.html

And also I think you can remove `PartialEq` if meet the related trouble.

Reply:
@AndreMouche Thx for pointing that out. I changed to only one RefCell. Pls review.